### PR TITLE
Add modified time to webkitbugspy.

### PR DESCRIPTION
**Pull request by bls1999**
#### 7a79cecafe05309d0d3308c479fd62be0827809c
<pre>
Add modified time to webkitbugspy.
<a href="https://bugs.webkit.org/show_bug.cgi?id=271272">https://bugs.webkit.org/show_bug.cgi?id=271272</a>
<a href="https://rdar.apple.com/125039020">rdar://125039020</a>

Reviewed by Jonathan Bedard.

Currently, webkitbugspy does not have a method to retrieve the last modified time of an Issue
object. This change adds this information into the Issue object such that it is callable.

Added getters/setters for the new modified property:
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/bugzilla.py
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/github.py
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/issue.py
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/radar.py

Added the new modified property into the mocks:
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/mocks/bugzilla.py
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/mocks/data.py
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/mocks/github.py
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/mocks/radar.py

Added new webkitbugspy unit tests:
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/tests/bugzilla_unittest.py
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/tests/github_unittest.py
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/tests/radar_unittest.py

Canonical link: <a href="https://commits.webkit.org/276428@main">https://commits.webkit.org/276428@main</a>
</pre>

https://github.com/WebKit/WebKit/commit/6117788364e9ee8cc5020fb57b15ee741c5ccfb4

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/44548 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/23625 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/46996 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/47199 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/40550 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/46853 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/27647 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/21015 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/36634 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/45124 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/20682 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/38339 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/17705 "Passed tests") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/44420 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/18118 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/39484 "Passed tests") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/2595 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/40743 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/39760 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/48824 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/19507 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/16040 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/43570 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/20847 "Built successfully") | | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/42322 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/9940 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/21175 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/20505 "Built successfully") | | |

### CODE REVIEW

**Comment by briannafan** [comment]
LGTM :)

**Comment by JonWBedard** [comment]
@bls1999 When I ran `test-webkitpy` locally, I got a bunch of failures. EWS seems to have thrown away the build request, but I suspect you'd see failures if you ran `test-webkitpy` locally.

**Comment by bls1999** [comment]
> @bls1999 When I ran `test-webkitpy` locally, I got a bunch of failures. EWS seems to have thrown away the build request, but I suspect you'd see failures if you ran `test-webkitpy` locally.

Looks like adding handling for `modified` not existing is what did the trick... was that the right move or should there be a different fix somewhere else to ensure it always exists?

The other odd thing is that the bugzilla mock didn't need the conditional. I added it anyway, just to be safe.

```
diff --git a/Tools/Scripts/libraries/webkitbugspy/webkitbugspy/mocks/bugzilla.py b/Tools/Scripts/libraries/webkitbugspy/webkitbugspy/mocks/bugzilla.py
index c52eebea8bef..3822b62bdcce 100644
--- a/Tools/Scripts/libraries/webkitbugspy/webkitbugspy/mocks/bugzilla.py
+++ b/Tools/Scripts/libraries/webkitbugspy/webkitbugspy/mocks/bugzilla.py
@@ -211,7 +211,7 @@ class Bugzilla(Base, mocks.Requests):
                 id=id,
                 summary=issue['title'],
                 creation_time=self.time_string(issue['timestamp']),
-                last_change_time=self.time_string(issue['modified']),
+                last_change_time=self.time_string(issue['modified' if issue.get('modified') else 'timestamp']),
                 status='REOPENED' if issue['opened'] else 'RESOLVED',
                 resolution='' if issue['opened'] else 'FIXED',
                 dupe_of=issue['original']['id'] if issue.get('original', None) else None,
diff --git a/Tools/Scripts/libraries/webkitbugspy/webkitbugspy/mocks/github.py b/Tools/Scripts/libraries/webkitbugspy/webkitbugspy/mocks/github.py
index 33739032c659..6bc768ed6b9c 100644
--- a/Tools/Scripts/libraries/webkitbugspy/webkitbugspy/mocks/github.py
+++ b/Tools/Scripts/libraries/webkitbugspy/webkitbugspy/mocks/github.py
@@ -139,7 +139,7 @@ class GitHub(Base, mocks.Requests):
             body=issue['description'],
             user=dict(login=self.users[issue['creator'].name].username),
             created_at=self.time_string(issue['timestamp']),
-            updated_at=self.time_string(issue['modified']),
+            updated_at=self.time_string(issue['modified' if issue.get('modified') else 'timestamp']),
             state='opened' if issue['opened'] else 'closed',
             labels=self._labels_for_issue(issue),
             milestone=dict(title=issue['milestone']) if issue.get('milestone') else None,
@@ -270,7 +270,7 @@ class GitHub(Base, mocks.Requests):
             body=issue['description'],
             user=dict(login=self.users[issue['creator'].name].username),
             created_at=self.time_string(issue['timestamp']),
-            updated_at=self.time_string(issue['modified']),
+            updated_at=self.time_string(issue['modified' if issue.get('modified') else 'timestamp']),
             state='opened' if issue['opened'] else 'closed',
             milestone=None,
             labels=self._labels_for_issue(issue),
@@ -300,7 +300,7 @@ class GitHub(Base, mocks.Requests):
             body=issue['description'],
             user=dict(login=self.users[issue['creator'].name].username),
             created_at=self.time_string(issue['timestamp']),
-            updated_at=self.time_string(issue['modified']),
+            updated_at=self.time_string(issue['modified' if issue.get('modified') else 'timestamp']),
             state='opened' if issue['opened'] else 'closed',
             labels=self._labels_for_issue(issue),
             assignee=dict(login=issue['assignee'].username),
diff --git a/Tools/Scripts/libraries/webkitbugspy/webkitbugspy/mocks/radar.py b/Tools/Scripts/libraries/webkitbugspy/webkitbugspy/mocks/radar.py
index 1f203d521cd0..ab0a30fee8b0 100644
--- a/Tools/Scripts/libraries/webkitbugspy/webkitbugspy/mocks/radar.py
+++ b/Tools/Scripts/libraries/webkitbugspy/webkitbugspy/mocks/radar.py
@@ -150,7 +150,7 @@ class RadarModel(object):
         self.id = issue['id']
         self.classification = issue.get('classification', 'Other Bug')
         self.createdAt = datetime.utcfromtimestamp(issue['timestamp'] - timedelta(hours=7).seconds)
-        self.lastModifiedAt = datetime.utcfromtimestamp(issue['modified'] - timedelta(hours=7).seconds)
+        self.lastModifiedAt = datetime.utcfromtimestamp(issue['modified' if issue.get('modified') else 'timestamp'] - timedelta(hours=7).seconds)
         self.assignee = self.Person(Radar.transform_user(issue['assignee']))
         self.description = self.CollectionProperty(self, self.DescriptionEntry(issue['description']))
         self.state = 'Analyze' if issue['opened'] else 'Verify'
```

File: Tools/Scripts/libraries/webkitbugspy/webkitbugspy/mocks/data.py [hunk 1]
```diff
@@ -49,6 +49,7 @@
     dict(
         title='Example issue 1',
         timestamp=1639536160,
+        modified=1710884407,
```
Comment (JonWBedard) [inline Tools/Scripts/libraries/webkitbugspy/webkitbugspy/mocks/data.py:52]: This is totally fine, although I might suggest making this number a little closer to the timestamp of the bug.

**Review by JonWBedard — APPROVED** [review]

File: Tools/Scripts/libraries/webkitbugspy/webkitbugspy/mocks/data.py (hunk 1 above)
Comment (bls1999) [reply Tools/Scripts/libraries/webkitbugspy/webkitbugspy/mocks/data.py:52]: Maybe silly, but for posterity, I always tend to use the current timestamp in my code when writing mock data. Kind of like a little easter egg for myself of, "oh hey, that's when I wrote that line!"

**Pull request merged**